src/vs/workbench/services/editor/common/editorGroupsService.ts

```diff
@@ -774,26 +774,16 @@ export interface IEditorGroup {
 	isActive(editor: EditorInput | IUntypedEditorInput): boolean;
 
 	/**
-	 * Selects the editor in the group. If active is set to true,
-	 * it will be the active editor in the group.
+	 * Selects the editors in the group.
+	 * If activateEditor is provided, it will become the active editor.
+	 * If activateEditor is not provided or not part of editors, the active editor will remain the same.
 	 */
-	selectEditor(editor: EditorInput, active?: boolean): Promise<void>;
+	selectEditors(editors: EditorInput[], activateEditor?: EditorInput): Promise<void>;
 
 	/**
-	 * Selects the editors in the group. If activeEditor is provided,
-	 * it will be the active editor in the group.
+	 * Unselects the editors in the group.
 	 */
-	selectEditors(editors: EditorInput[], activeEditor?: EditorInput): Promise<void>;
-
-	/**
-	 * Unselects the editor in the group. If the editor is not specified, unselects the active editor.
-	 */
-	unSelectEditor(editor: EditorInput): Promise<void>;
-
-	/**
-	 * Unselects the editors in the group. If the editor is not specified, unselects the active editor.
-	 */
-	unSelectEditors(editors: EditorInput[]): Promise<void>;
+	unselectEditors(editors: EditorInput[]): Promise<void>;
```

Comment:
The reason I've not done this approach is because this moves all the work to the consumer. Also it seems a bit weird to have API where the consumer needs to know that the first item in the array will become active. 
I'll think about this

Reply:
Lets move that discussion out of the PR.